src/main/scala/performanceanalysis/administrator/AdministratorActor.scala

```diff
@@ -15,21 +15,27 @@ object AdministratorActor {
   def props(logReceiverActor: ActorRef): Props = Props(new AdministratorActor(logReceiverActor))
 }
 
-class AdministratorActor(logReceiverActor: ActorRef) extends Actor with ActorLogging with LogParserActorCreater {
-  this: LogParserActorCreater =>
+class AdministratorActor(logReceiverActor: ActorRef) extends Actor with ActorLogging with LogParserActorCreator {
+  this: LogParserActorCreator =>
 
   def receive: Receive = normal(Map.empty[String, ActorRef])
 
   def normal(logParserActors: Map[String, ActorRef]): Receive = {
     case RegisterComponent(componentId) =>
-      handleRegisterComponent(logParserActors, componentId, sender)
+      handleRegisterComponent(logParserActors, componentId)
     case GetDetails(componentId) =>
-      handleGetDetails(logParserActors, componentId, sender)
+      handleGetDetails(logParserActors, componentId)
     case GetRegisteredComponents =>
-      ???
+      handleRegisteredComponents(logParserActors)
   }
 
-  private def handleRegisterComponent(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
+  private def handleRegisteredComponents(logParserActors: Map[String, ActorRef]) = {
```

Comment:
Wouldn't the name ```handleGetRegisteredComponents``` be more descriptive than ```handleRegisteredComponents```? And you forgot to add unit tests for this.